app/utils/draft/index.ts

```diff
@@ -33,3 +47,240 @@ export function alertErrorWithFallback(intl: IntlShape, error: any, fallback: Me
     }
     Alert.alert('', msg, buttons);
 }
+
+export function alertAttachmentFail(intl: IntlShape, accept: AlertCallback, cancel: AlertCallback) {
+    Alert.alert(
+        intl.formatMessage({
+            id: 'mobile.post_textbox.uploadFailedTitle',
+            defaultMessage: 'Attachment failure',
+        }),
+        intl.formatMessage({
+            id: 'mobile.post_textbox.uploadFailedDesc',
+            defaultMessage: 'Some attachments failed to upload to the server. Are you sure you want to post the message?',
+        }),
+        [{
+            text: intl.formatMessage({id: 'mobile.channel_info.alertNo', defaultMessage: 'No'}),
+            onPress: cancel,
+        }, {
+            text: intl.formatMessage({id: 'mobile.channel_info.alertYes', defaultMessage: 'Yes'}),
+            onPress: accept,
+        }],
+    );
+}
+
+export const textContainsAtAllAtChannel = (text: string) => {
+    const textWithoutCode = text.replace(CODE_REGEX, '');
+    return (/(?:\B|\b_+)@(channel|all)(?!(\.|-|_)*[^\W_])/i).test(textWithoutCode);
+};
+
+export const textContainsAtHere = (text: string) => {
+    const textWithoutCode = text.replace(CODE_REGEX, '');
+    return (/(?:\B|\b_+)@(here)(?!(\.|-|_)*[^\W_])/i).test(textWithoutCode);
+};
+
+export const groupsMentionedInText = (groupsWithAllowReference: Group[], text: string) => {
+    if (!groupsWithAllowReference.length) {
+        return [];
+    }
+    const textWithoutCode = text.replace(CODE_REGEX, '');
+    const mentions = textWithoutCode.match(AT_MENTION_REGEX_GLOBAL) || [];
+    return groupsWithAllowReference.filter((g) => mentions.includes(g.id));
+};
+
+export const mapGroupMentions = (channelMemberCountsByGroup: ChannelMemberCountByGroup[], groupMentions: Group[]) => {
+    let memberNotifyCount = 0;
+    let channelTimezoneCount = 0;
+    const groupMentionsSet = new Set<string>();
+    const mappedChannelMemberCountsByGroup: ChannelMemberCountsByGroup = {};
+    channelMemberCountsByGroup.forEach((group) => {
+        mappedChannelMemberCountsByGroup[group.group_id] = group;
+    });
+    groupMentions.
+        forEach((group) => {
+            const mappedValue = mappedChannelMemberCountsByGroup[group.id];
+            if (mappedValue?.channel_member_count > NOTIFY_ALL_MEMBERS && mappedValue?.channel_member_count > memberNotifyCount) {
+                memberNotifyCount = mappedValue.channel_member_count;
+                channelTimezoneCount = mappedValue.channel_member_timezones_count;
+            }
+            groupMentionsSet.add(`@${group.name}`);
```

Comment:
I had an issue with groups having no `name`, making the `@${group.name} == '@'` maybe something to check against here as well.